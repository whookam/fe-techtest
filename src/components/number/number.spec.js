import React from 'react';
import TestRenderer from 'react-test-renderer';
import { render} from '@testing-library/react';
import 'jest-styled-components';

import NumberBox from './index';

describe('Number box component', () => {
  const { getByTestId } = render(<NumberBox number={{number: 10, multiple: false}} />);

  it('displays the correct content when given 10 as a number', () => {
    const displayNumnber = getByTestId('displayNumber');
    expect(displayNumnber.innerHTML).toEqual("10");
  });

  test('Render snapshot test of the numberBox component', () => {
    const renderedNumberBox =  TestRenderer.create(<NumberBox number={{number: 10, multiple: false}} />).toJSON();
    expect(renderedNumberBox).toMatchSnapshot();
  });

  test('it renderes the correct CSS styles for the NumberBox component when inactive', () => {
    const { getByTestId } = render(<NumberBox number={{number: 15, multiple: false, rootNumber: false}} />);
    const content = getByTestId('displayNumber');
    expect(content).toHaveStyleRule('background', 'white');
    expect(content).toHaveStyleRule('border-color', '#333');
  });

  test('it renderes the correct CSS styles for the NumberBox component when active', () => {
    const { getByTestId } = render(<NumberBox number={{number: 10, multiple: true, rootNumber: false}} />);
    const content = getByTestId('displayNumber');
    expect(content).toHaveStyleRule('background', 'lightblue');
    expect(content).toHaveStyleRule('border-color', '#333');
  });
  
  test('it renderes the correct CSS styles for the NumberBox component when the root number', () => {
    const { getByTestId } = render(<NumberBox number={{number: 4, multiple: true, rootNumber: true}} />);
    const content = getByTestId('displayNumber');
    expect(content).toHaveStyleRule('background', 'lightblue');
    expect(content).toHaveStyleRule('border-color', 'yellow');
  });

});
