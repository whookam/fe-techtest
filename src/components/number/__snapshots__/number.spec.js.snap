// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Number box component Render snapshot test of the numberBox component 1`] = `
.c0 {
  display: inline-block;
  width: 30%;
  height: 250px;
  border-style: solid;
  border-width: 3px;
  border-color: #333;
  border-radius: 10px;
  box-shadow: 2px 2px 10px #999;
  font-family: 'Helvetica Neue, Segoe, Verdana, sans-serif';
  margin: 1%;
  text-align: center;
  line-height: 250px;
  font-size: 72px;
  background: white;
}

.c0:hover {
  background: #FACACE;
  cursor: pointer;
}

.c0:active,
.c0:focus {
  background: #FAF8B1;
}

@media only screen and (max-width:768px) {
  .c0 {
    width: 46%;
  }
}

@media only screen and (max-width:600px) {
  .c0 {
    width: 98%;
    margin: 20px 0;
  }
}

<div
  className="c0"
  data-testid="displayNumber"
  onClick={[Function]}
>
  10
</div>
`;
